### PR TITLE
Avoid unary minus on np.sum in the sampled softmax loss reference

### LINKED ISSUE #127559: PyLint presubmit fails with false-positive E1130 on nn_test.py (numpy 2.0.2 + pylint 2.13.9)
### Issue type

Bug (CI tooling)

### Source

`pylint-presubmit.yml` (the `PyLint` check) on any PR touching `tensorflow/python/ops/nn_test.py`

### Description

The PyLint presubmit fails with two `E1130` reports that point at **pre-existing upstream code**, not at the contributor's own change:

```
tensorflow/python/ops/nn_test.py:879:13: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
tensorflow/python/ops/nn_test.py:936:13: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

Both lines are `return -np.sum(targets * np.log(pred + 1.0e-20), axis=1)`, inside the sampled-softmax-loss reference implementations used by `testSampledSoftmaxLoss` and `testSampledSoftmaxLossBf16`.

Because the workflow runs pylint over the entire changed file, **every PR that modifies `nn_test.py` inherits this failure**.

### Root cause

`.github/workflows/pylint-presubmit.yml` pins `pylint==2.13.9` but leaves `numpy` unpinned:

```
pip install pylint==2.13.9 numpy wheel
```

The job runs on Python 3.9, where unpinned `numpy` resolves to **2.0.2** (the last release that supports 3.9). astroid 2.11.7 (the inference engine behind pylint 2.13.9) fails to resolve `np.sum` under numpy 2.0.2 and infers it as `NoneType`, so the unary minus is reported as an invalid operand.

Verified locally with pylint 2.13.9 and the repository's own `tensorflow/tools/ci_build/pylintrc`:

| numpy | result |
|---|---|
| 2.0.2 | 9.93/10 — exactly these two `E1130` reports (exit 2) |
| 2.5.3 | 10.00/10 — no reports |

The Python version is not a factor: the same split reproduces on Python 3.12, which isolates numpy as the trigger.

### Suggested fix

Either:

1. Rewrite the two expressions as `-1.0 * np.sum(...)`, which I verified restores 10.00/10 under numpy 2.0.2 with no behaviour change; or
2. Pin `numpy` in the workflow (or upgrade the pinned pylint) so that inference works again.

I am happy to send the one-line fix if that is preferred.

### PR DESCRIPTION
**Pull request by Phoebus-Liu**
Fixes #127559.

The PyLint presubmit currently fails for every PR that touches
`tensorflow/python/ops/nn_test.py`, with two false positives:

    tensorflow/python/ops/nn_test.py:879:13: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
    tensorflow/python/ops/nn_test.py:936:13: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)

Both lines are pre-existing code: `return -np.sum(targets * np.log(pred + 1.0e-20), axis=1)` in the sampled softmax loss reference implementations used by `testSampledSoftmaxLoss` and `testSampledSoftmaxLossBf16`.

**Root cause.** `.github/workflows/pylint-presubmit.yml` pins `pylint==2.13.9` but leaves `numpy` unpinned. The job runs on Python 3.9, where `numpy` resolves to 2.0.2; astroid 2.11.7 cannot resolve `np.sum` under that version and infers it as `NoneType`, so the unary minus is reported as an invalid operand. This is a tooling artefact rather than a real defect.

**Fix.** Express the negation with `np.negative()` instead of the unary minus. It is behaviour-preserving and keeps the intent explicit.

Note that `np.negative` is a faithful replacement for the unary minus in a way that multiplying by `-1.0` is not: for integral input, `-np.sum(x)` and `np.negative(np.sum(x))` both stay int64, while `-1.0 * np.sum(x)` promotes to float64. At this call site `targets * np.log(...)` is always floating point, so all three forms agree bit for bit — but `np.negative` preserves dtype in general.

**Verification.**

- `pylint --rcfile=tensorflow/tools/ci_build/pylintrc tensorflow/python/ops/nn_test.py` with pylint 2.13.9:
  - numpy 2.0.2 (what CI resolves to on Python 3.9) → **10.00/10**; previously 9.93/10 with two `E1130` reports
  - numpy 2.5.3 → 10.00/10
- `-np.sum(x)`, `np.negative(np.sum(x))` and `-1.0 * np.sum(x)` compared over float16/32/64 and int32/64 inputs across several shapes: identical values in all cases, and `np.negative` additionally preserves dtype.
- `ComputeSampledLogitsTest.testSampledSoftmaxLoss` and `testSampledSoftmaxLossBf16` pass on TensorFlow 2.21.

This also unblocks #127550, which touches the same file and currently inherits the failure.

### CODE REVIEW

File: tensorflow/python/ops/nn_test.py [hunk 1]
```diff
@@ -876,7 +876,7 @@ def _SoftmaxCrossEntropyWithLogits(logits, targets):
       stable_exp_logits = np.exp(logits -
                                  np.amax(logits, axis=1, keepdims=True))
       pred = stable_exp_logits / np.sum(stable_exp_logits, 1, keepdims=True)
-      return -np.sum(targets * np.log(pred + 1.0e-20), axis=1)
+      return -1.0 * np.sum(targets * np.log(pred + 1.0e-20), axis=1)
```
[the remark this answers is not in the document]
Comment (Phoebus-Liu) [reply tensorflow/python/ops/nn_test.py]: Good suggestion — switched to `np.negative()` in 6ebf4a3a (both occurrences).

I checked all three forms before changing anything:

- **pylint 2.13.9 + the repo's own rcfile, numpy 2.0.2** (what CI resolves to on Python 3.9): `-np.sum(...)` -> 9.93/10 with exactly these two `E1130` reports; `np.negative(...)` and `-1.0 * ...` both -> 10.00/10. Same result with numpy 2.5.3.
- **`np.negative` is also the more faithful replacement.** For integral input it preserves dtype (`-np.sum(x)` and `np.negative(np.sum(x))` both stay int64), while `-1.0 * np.sum(x)` promotes to float64. At this call site `targets * np.log(...)` is always floating point, so all three agree bit for bit — but `np.negative` matches what the unary minus did, in every dtype.
- `ComputeSampledLogitsTest.testSampledSoftmaxLoss` and `testSampledSoftmaxLossBf16` pass on TF 2.21 with the new form.

File: tensorflow/python/ops/nn_test.py [hunk 2]
```diff
@@ -933,7 +933,7 @@ def _SoftmaxCrossEntropyWithLogits(logits, targets):
...
```
[the remark this answers is not in the document]
Comment (Phoebus-Liu) [reply tensorflow/python/ops/nn_test.py]: Done in 6ebf4a3a as well. See the sibling comment for the measurements: `np.negative` clears the E1130 report under numpy 2.0.2 just as `-1.0 *` did, and unlike the multiplication it preserves dtype for integral input.

**Review by dmiltr3 — APPROVED** [review]
The code changes look good and internal presubmit test suites have successfully passed. Approving.

**Review by dmiltr3 — APPROVED** [review]
Thanks for fixing the false positive with PyLint! The changes look good and all internal presubmit tests have passed successfully. Approving.

**Comment by Phoebus-Liu** [comment]
Thank you, @dmiltr3! I'm glad the PyLint false positive is resolved, and I appreciate your approval. Switching to `np.negative()` keeps the reference computation equivalent while avoiding the unary-minus false positive. Thanks again for your review.
